test/e2e/run.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash -e
+
+SUPPORTED_REGISTRY=("zot" "dockerhub")
+
+function help {
+    echo "Usage"
+    echo "  run.sh <registry-name> <notation-binary-path> [old-notation-binary-path]"
+    echo ""
+    echo "Arguments"
+    echo "  registry-name             is the registry to use for running the E2E test. Currently support: ${SUPPORTED_REGISTRY[@]}."
+    echo "  notation-binary-path      is the path of the notation executable binary file."
+    echo "  old-notation-binary-path  is the path of an old notation executable bianry file. If it is not set, an RC.1 Notation will be downloaded automatically."
+}
+
+# check registry name
+REGISTRY_NAME=$1
+if [ -z "$REGISTRY_NAME" ]; then
+    echo "registry name is missing."
+    help
+    exit 1
+fi
+
+# check notation binary path.
+export NOTATION_E2E_BINARY_PATH=$(if [ ! -z "$2" ]; then realpath $2; fi)
+if [ ! -f "$NOTATION_E2E_BINARY_PATH" ];then
+    echo "notation binary path doesn't exist."
+    help
+    exit 1
+fi
+
+# check old notation binary path for forward compatibility test.
+export NOTATION_E2E_OLD_BINARY_PATH=$(if [ ! -z "$3" ]; then realpath $3; fi)
+if [ ! -f "$NOTATION_E2E_OLD_BINARY_PATH" ];then
+    OLD_NOTATION_DIR=/tmp/notation_old
+    export NOTATION_E2E_OLD_BINARY_PATH=$OLD_NOTATION_DIR/notation
+    mkdir -p $OLD_NOTATION_DIR
+
+    echo "Old notation binary path doesn't exist."
+    echo "Try to use old notation binary at $NOTATION_E2E_OLD_BINARY_PATH"
+
+    if [ ! -f $NOTATION_E2E_OLD_BINARY_PATH ]; then
+        TAG=1.0.0-rc.1 # without 'v'
+        echo "Didn't find old notation binary locally. Try to download notation v$TAG."
+
+        TAR_NAME=notation_${TAG}_linux_amd64.tar.gz
+        URL=https://github.com/notaryproject/notation/releases/download/v${TAG}/$TAR_NAME
+        wget $URL -P $OLD_NOTATION_DIR
+        tar -xf $OLD_NOTATION_DIR/$TAR_NAME -C $OLD_NOTATION_DIR
+
+        if [ ! -f $NOTATION_E2E_OLD_BINARY_PATH ]; then
+            echo "Failed to download old notation binary for forward compatibility test."
+            exit 1
+        fi
+        echo "Downloaded notation v$TAG at $NOTATION_E2E_OLD_BINARY_PATH"
+    fi
+fi
+
+# install dependency
+go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo@v2.3.0
+
+# setup registry
+case $REGISTRY_NAME in
+
+    "zot")
+        source ./scripts/zot.sh
+        ;;
+
+    "dockerhub")
+        source ./scripts/dockerhub.sh
+        ;;
+
```

Comment:
is this working? I thought dockerhub doesn't support OCI1.1 spec

Reply:
Yes, it works! Dockerhub support OCI artifact now.